src/gluonts/nursery/duru/duru/vdvae_conv.py

```diff
@@ -0,0 +1,950 @@
+        mu_p_z, log_sigma_p_z, z_iplus1_0_sigma = (
+            p_z_activations[:, : self.H.z_channels, ...],
+            p_z_activations[:, self.H.z_channels : self.H.z_channels * 2, ...],
+            p_z_activations[:, self.H.z_channels * 2 :, ...],
+        )
+        scale = get_stable_scale(log_sigma_p_z)
+        scale = scale * temp
+        p_z = D.Independent(D.Normal(loc=mu_p_z, scale=scale), 2)
+
+        # state
+        state = state + z_iplus1_0_sigma
+        if set_z_sample is not None and not self.H.conditional:
+            z_sample_uncond = set_z_sample
+        elif set_z_sample is None and not self.H.conditional:
+            z_sample_uncond = (
+                p_z.rsample()
+            )  # properly taking a sample, not the mean
+        elif set_z_sample is None and self.H.conditional:
+            z_sample_uncond = p_z.rsample(
+                torch.Size([])
+            )  # note: only one sample is drawn, because "sample_shape is part of the batch_shape"
+            # z_sample_uncond = torch.permute(z_sample_uncond, (1, 0, 2, 3))  # reorder dimensions s.t. (batch_size, sample_shape, event_shape_1, event_shape_2)
+            # z_sample_uncond = torch.flatten(z_sample_uncond, start_dim=0, end_dim=1)   # dimensions: (batch_size * sample_shape, event_shape_1, event_shape_2), s.t. in 0th dimension, blocks of sample_shape each correponding to one datapoint in x_context (in Decoder)
+        state = state + self.sample_proj_block(z_sample_uncond)
+        state = self.mean_block(state)
+
+        return state, z_sample_uncond
+
+
+class UpInterpolate(nn.Module):
+    def __init__(self, up_rate):
+        super().__init__()
+        # F.interpolate(xs[self.mixin][:, :x.shape[1], ...], scale_factor=self.base // self.mixin)
+        self.up_rate = up_rate
+
+    def forward(self, x):
+        x = F.interpolate(x, scale_factor=self.up_rate)
+
+        return x
+
+
+class Bias_xs(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.param_names = []
+
+
+class Decoder(nn.Module):
+    def __init__(self, H):
+        super().__init__()
+        self.H = H
+
+        self.blocks = torch.nn.ModuleList()
+        if H.likelihood_type == "GaussianSigmaHyperparam":
+            self.out_conv = get_conv_layer(
+                in_channels=self.H.vdvae_dec_state_channels,
+                out_channels=self.H.n_meas,
+                kernel_size=1,
+                stride=1,
+                padding=0,
+            )
+        elif H.likelihood_type == "GaussianSigmaEstimated":
+            self.out_conv = get_conv_layer(
+                in_channels=self.H.vdvae_dec_state_channels,
+                out_channels=self.H.n_meas * 2,
+                kernel_size=1,
+                stride=1,
+                padding=0,
+            )
+
+        self.layer_index_to_out_res = (
+            {}
+        )  # mapping from layer index to resolution of the output of that layer
```

Comment:
Let's have these comments above the line they are describing to avoid these breaks.

Reply:
done